src/vmm/src/vmm_config/net.rs

```diff
@@ -147,6 +147,21 @@ impl NetBuilder {
         self.net_devices.iter_mut()
     }
 
+    /// Adds an existing network device in the builder.
+    pub fn add_device(&mut self, device: Arc<Mutex<Net>>) {
+        if let Some(index) = self
+            .net_devices
+            .iter()
+            .position(|net| {
+                net.lock().expect("Poisoned lock").id() == &device.lock().expect("Poisoned lock").id
+            })
+        {
+            self.net_devices.swap_remove(index);
+        }
```

Comment:
When restoring from a snapshot there will never be two net devices with the same id. The Firecracker VM on which the snapshot was taken would have prevented this at net device creation time.